internal/services/disbursement_management_service.go

```diff
@@ -241,20 +240,16 @@ func (s *DisbursementManagementService) StartDisbursement(ctx context.Context, d
 			}
 
 			// 4. Check if there is enough balance from the distribution wallet for this disbursement along with any pending disbursements
-			rootAccount, err := s.horizonClient.AccountDetail(
-				horizonclient.AccountRequest{AccountID: distributionPubKey})
+			var availableBalance float64
+			balances, err := s.distributionAccountService.GetBalances(ctx, distributionAccount)
 			if err != nil {
-				err = tssUtils.NewHorizonErrorWrapper(err)
-				return nil, fmt.Errorf("cannot get details for root account from horizon client: %w", err)
+				return nil, fmt.Errorf("cannot get distribution account balances: %w", err)
 			}
 
-			var availableBalance float64
-			for _, b := range rootAccount.Balances {
-				if disbursement.Asset.EqualsHorizonAsset(b.Asset) {
-					availableBalance, err = strconv.ParseFloat(b.Balance, 64)
-					if err != nil {
-						return nil, fmt.Errorf("cannot convert Horizon distribution account balance %s into float: %w", b.Balance, err)
-					}
+			for asset, b := range balances {
+				if disbursement.Asset.EqualsDistributionAccountMapID(asset) {
+					availableBalance = b
+					break
```

Comment:
Why don't you use GetBalance directly, rather than using GetBalances and finding it here?